### PR TITLE
Bug/proprietary binary obeys `always_download`

### PR DESCRIPTION
**Pull request by robertmaynard**
## Description
Update rapids_cpm_nvcomp to check for a local version before downloading the properitary version

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/rapids-cmake/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [x] The `cmake-format.json` is up to date with these changes.

### CODE REVIEW

**Review by vyasr — APPROVED** [review]
One small request for extra docs, otherwise looks good. Feel free to merge once you've addressed the doc request however you see fit.

File: rapids-cmake/cpm/detail/package_details.cmake [hunk 1]
```diff
@@ -99,6 +99,7 @@ function(rapids_cpm_package_details package_name version_var url_var tag_var sha
   set(${shallow_var} ${git_shallow} PARENT_SCOPE)
   set(${exclude_from_all_var} ${exclude_from_all} PARENT_SCOPE)
   if(DEFINED always_download)
+    set(rapids_cmake_always_download ${always_download} PARENT_SCOPE)
```
[the remark this answers is not in the document]
Comment (robertmaynard) [reply rapids-cmake/cpm/detail/package_details.cmake:107]: Documented

**Comment by bdice** [comment]
@robertmaynard @vyasr Thanks for seeing this through!

**Pull request merged**